### ISSUE TITLE
[PHP 7.1] Out of memory Exception when using opcache

### ISSUE THREAD

**Issue by MatthiasKuehneEllerhold** [oerdnj/deb.sury.org#507]
After upgrading our (debian jessie) docker images to PHP 7.1 we have a reproducible out of memory exception when opcache is enabled:
```
2016-12-12T10:30:39+01:00 ERR (3): ErrorException: Out of memory (allocated 2097152) (tried to allocate 280399987992672 bytes) in /XXX/vendor/zendframework/zend-escaper/src/Escaper.php:144 in /XXX/Logger.php:86
Stack trace:
#0 [internal function]: XXX\Logger::XXX\{closure}()
#1 {main}
```
The call in question is a call to the zend-escaper with a 20 character string. The same code works flawlessly:
* if we put `opcache_reset();` as the first statement in our app entry-point (`index.php`)
* if we debug with xDebug (I guess Opcache is disabled during the debugging?)
* if I use another docker container with dotdeb's PHP 7.0.13 & 7.0.14
* if I use another docker container with sury's PHP 7.0.14 (`PHP 7.0.14-1+0~20161212122340.19+jessie~1.gbp8dc455` to be exact)

(The same problem persists in a Debian Jessie VM, so its not docker-specific.)

I dont know if this is a problem of PHP 7.1.0 in general or if its some problem of your repository. I searched around and found no similar bug report. (see https://bugs.php.net/search.php?limit=30&order_by=id&direction=DESC&cmd=display&status=Open&bug_type=All&phpver=7.1)

Has anybody the same problems with the 7.1 upgrade? 
Is there some fundamentally different between dotdeb's and sury's PHP packages?

Versions:
```
PHP 7.1.0-2+0~20161206214730.10+jessie~1.gbp2889f0 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.1.0-dev, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.1.0-2+0~20161206214730.10+jessie~1.gbp2889f0, Copyright (c) 1999-2016, by Zend Technologies
    with Xdebug v2.5.0, Copyright (c) 2002-2016, by Derick Rethans
```

Let me know if you want more informations regarding this.

**Comment by oerdnj** [comment]
There are no packaging modifications that should cause such behavior, so I would recommend filling this with upstream.

dotdeb doesn't have PHP 7.1, so there's no point in comparison, unless you can reproduce it with PHP 7.0.x from me.

Please report upstream bug URL here, once you report it there.

**Comment by MatthiasKuehneEllerhold** [comment]
Thank you!

**Comment by MatthiasKuehneEllerhold** [comment]
I managed to build a minimal version of the test case that breaks the opcache: https://github.com/MatthiasKuehneEllerhold/php71-opcache-failure/blob/master/test.php
and opened a new bug on https://bugs.php.net/bug.php?id=73801

**Comment by MatthiasKuehneEllerhold** [comment]
This bug will be fixed in PHP 7.1.1.

**Comment by oerdnj** [comment]
I have pulled the upstream fixes and uploaded it as `php7.1_7.1.0-4...`

**Comment by MatthiasKuehneEllerhold** [comment]
Did you upload jessie too? ;)

**Comment by MatthiasKuehneEllerhold** [comment]
Found it!

**Comment by oerdnj** [comment]
JFTR there's a debian-jenkins-glue builder and you can check it at https://howl.labs.nic.cz/jenkins/ (most jobs have artifacts that can be checked).

**Comment by MatthiasKuehneEllerhold** [comment]
Sorry, I still got the same problem with `PHP Version 7.1.0-4+0~20161222090610.12+jessie~1.gbpbfa2ce`

**Comment by MatthiasKuehneEllerhold** [comment]
nikic@php.net just wrote:
> Just checked, your test case runs fine under current 7.1 branch. Note that there have been three different fixes regarding handling of return types in opcache, namely:
>
> https://github.com/php/php-src/commit/3b79f8f408ab090825bc15656e517746fdc43db9
> https://github.com/php/php-src/commit/60249f439012598717f97ce250f06181116502ea 
> https://github.com/php/php-src/commit/6bd41a1d47417ad2c7938a57c8abca4e210d5cf6
>
> I did not check it, but it may be that only pulling in the last one does not fix the issue.

Could you please confirm that you've pulled all 3 commits into the 7.1.0-4? Sorry for bothering you so much right before the holidays... ^^

**Comment by oerdnj** [comment]
Ah, that would be it. I pulled only the one referenced from the bug.

**Comment by oerdnj** [comment]
I missed the middle patch because it was not references in NEWS. -5 build should have all three patches.

**Comment by MatthiasKuehneEllerhold** [comment]
I'll have to check that after the holidays. Thanks for your effort!

**Comment by MatthiasKuehneEllerhold** [comment]
Happy new year everybody!

Sadly new year, same bug. I've got the same problem with `PHP 7.1.0-5+0~20161222133327.13+jessie~1.gbp46a191` :/

**Comment by MatthiasKuehneEllerhold** [comment]
@oerdnj Is there anything you can do or should I just wait for 7.1.1 ?

**Comment by oerdnj** [comment]
I can do a RC1 build in php-qa ppa if that helps

**Comment by oerdnj** [comment]
I can do a RC1 build in php-qa ppa if that helps

**Comment by MatthiasKuehneEllerhold** [comment]
No, thats not necessary. Thank you though, I'll wait for 7.1.1.

**Comment by MatthiasKuehneEllerhold** [comment]
7.1.1 is out: http://www.php.net/ChangeLog-7.php#7.1.1 lets hope it works!

**Comment by MatthiasKuehneEllerhold** [comment]
Ok, I can confirm that the error is fixed in `PHP 7.1.1-1+0~20170120094658.14+jessie~1.gbp69d416`.

Thanks!
